service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -362,24 +365,70 @@ public void createReferenceResource(ReferencedResource resource)
     storeResource(resource);
   }
 
+  @WriteTransaction
+  private boolean updateResource(
```

Comment:
You cannot put transaction control on a `private` method. The way it works is Spring inserts a layer between the DAO caller and the DAO. That layer processes the annotation and starts transactions.

The general guideline for our DAOs is to put the transaction control on public interface methods. If there are shared methods, make them internal and private. So you would not have one public transactional method call another public transaction method. If that doesn't make sense, we can talk 1:1.